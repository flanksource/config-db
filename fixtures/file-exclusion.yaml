apiVersion: configs.flanksource.com/v1
kind: ScrapeConfig
metadata:
  name: exclude-file-secrets
spec:
  file:
    - type: $.name
      id: $.id
      name: $.name
      class: MySecrets
      transform:
        script:
          javascript: |+
            for (var i = 0; i < config.length; i++) {
              config[i].id = i + 1
            }
            JSON.stringify(config)
        exclude:
          - jsonpath: '.password'
          - types:
              - Config2
            jsonpath: '.secret'
      paths:
        - fixtures/data/multiple-configs.json
