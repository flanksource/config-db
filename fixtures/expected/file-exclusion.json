[
  {
    "last_modified": "0001-01-01T00:00:00Z",
    "config_type": "Config1",
    "config_class": "MySecrets",
    "name": "Config1",
    "id": "1",
    "config": {
      "id": 1,
      "name": "Config1",
      "secret": "secret_1"
    }
  },
  {
    "last_modified": "0001-01-01T00:00:00Z",
    "config_type": "Config2",
    "config_class": "MySecrets",
    "name": "Config2",
    "id": "2",
    "config": {
      "id": 2,
      "name": "Config2"
    }
  }
]
